src/GPUInfoProvider.cpp

```diff
@@ -0,0 +1,284 @@
+        {
+            {"size", physicalDeviceInfo.memoryProperties.memoryHeaps[i].size},
+            {"flags", lug::Graphics::Vulkan::API::RTTI::VkMemoryHeapFlagsToStrVec(physicalDeviceInfo.memoryProperties.memoryHeaps[i].flags)},
+        }
+        );
+    }
+
+    for (auto tmp : physicalDeviceInfo.queueFamilies) {
+        json["queues"].push_back({
+            {"queueFlags", lug::Graphics::Vulkan::API::RTTI::VkQueueFlagsToStrVec(tmp.queueFlags)},
+            {"queueCount", tmp.queueCount},
+            {"timestampValidBits", tmp.timestampValidBits},
+            {"minImageTransferGranularity", {
+                {"width", tmp.minImageTransferGranularity.width},
+                {"height", tmp.minImageTransferGranularity.height},
+                {"depth", tmp.minImageTransferGranularity.depth}
+            }}
+        });
+    }
+
+    for (auto extension : physicalDeviceInfo.extensions) {
+        json["extension"].push_back(
+        {
+            {"extensionName", extension.extensionName},
+            {"specVersion", {
+                {"major", static_cast<uint32_t>(lug::Core::Version::fromInt(extension.specVersion).major)},
+                {"minor", static_cast<uint32_t>(lug::Core::Version::fromInt(extension.specVersion).minor)},
+                {"patch", static_cast<uint32_t>(lug::Core::Version::fromInt(extension.specVersion).patch)}
+            }},
+        }
+        );
+    }
+
+    for (auto format : physicalDeviceInfo.formatProperties) {
+        json["formats"].push_back(
+        {
+            lug::Graphics::Vulkan::API::RTTI::toStr(format.first),
+            {
+                {"linearTilingFeatures", lug::Graphics::Vulkan::API::RTTI::VkFormatFeatureFlagsToStrVec(format.second.linearTilingFeatures)},
+                {"optimalTilingFeatures", lug::Graphics::Vulkan::API::RTTI::VkFormatFeatureFlagsToStrVec(format.second.optimalTilingFeatures)},
+                {"bufferFeatures", lug::Graphics::Vulkan::API::RTTI::VkFormatFeatureFlagsToStrVec(format.second.bufferFeatures)},
+            }
+        }
+        );
+    }
+
+    json["zSwapChainInfo"].push_back(
+    {
+        {"minImageCount", physicalDeviceInfo.swapchain.capabilities.minImageCount},
+        {"maxImageCount", physicalDeviceInfo.swapchain.capabilities.maxImageCount},
+        {"currentExtent", {
+            {"width", physicalDeviceInfo.swapchain.capabilities.currentExtent.width},
+            {"height", physicalDeviceInfo.swapchain.capabilities.currentExtent.height}
+        }
+        },
+        {"minImageExtent",{
+            {"width", physicalDeviceInfo.swapchain.capabilities.minImageExtent.width},
+            {"height", physicalDeviceInfo.swapchain.capabilities.minImageExtent.height}
+        }
+        },
+        {"maxImageExtent", {
+            {"width", physicalDeviceInfo.swapchain.capabilities.maxImageExtent.width},
+            {"height", physicalDeviceInfo.swapchain.capabilities.maxImageExtent.height}
+        }
```

Comment:
Because you are opening two brackets on the same line, you need the two closing ones on the same line too.